packages/react-components/src/components/OnboardingChecklist/OnboardingChecklist.tsx

```diff
@@ -0,0 +1,184 @@
+  const [isOpen, setIsOpen] = React.useState(true);
+  const [currentContainerHeight, setCurrentContainerHeight] = React.useState<
+    number | undefined
+  >(undefined);
+  const containerRef = React.useRef<HTMLDivElement>(null);
+
+  const handleButtonClick = () => {
+    setIsOpen((prev) => !prev);
+  };
+
+  React.useEffect(() => {
+    const delay = completeItem?.delay || 1500;
+    const container = containerRef.current;
+
+    if (isCompleted && container) {
+      const currentHeight = container.offsetHeight;
+      setCurrentContainerHeight(currentHeight);
+
+      setTimeout(() => {
+        setComplete(true);
+        setIsOpen(false);
+      }, delay);
+    }
+  }, [isCompleted]);
+
+  return (
+    <div
+      ref={containerRef}
+      className={cx(
+        styles[baseClass],
+        {
+          [styles[`${baseClass}--complete`]]: complete && !isOpen,
+        },
+        className
+      )}
+      style={{
+        height:
+          !complete || isOpen
+            ? currentContainerHeight
+            : COMPLETE_CONTAINER_EIGHT,
+      }}
+    >
+      {isOpen && (
+        <>
+          <div className={styles[`${baseClass}__column`]}>
+            <div className={styles[`${baseClass}__header`]}>
+              {titleLabel && (
+                <Text
+                  size="lg"
+                  className={styles[`${baseClass}__header__label`]}
+                >
+                  {titleLabel}
+                </Text>
+              )}
+              <Heading
+                size="lg"
+                className={styles[`${baseClass}__header__title`]}
+              >
+                {title}
+              </Heading>
+            </div>
+            <div className={styles[`${baseClass}__checklist`]}>
+              {items.map((item, index) => (
+                <CheckListItem
+                  key={index}
+                  id={item.id}
+                  description={item.description}
+                  title={item.title}
+                  isActive={item.id === activeId}
+                  isChecked={checkedId.includes(item.id)}
+                  isLastElement={index === items.length - 1}
+                  onClick={onActiveChange}
+                  cta={item.cta}
+                />
+              ))}
+            </div>
+          </div>
+          <div
+            className={cx(
+              styles[`${baseClass}__column`],
+              styles[`${baseClass}__column--right`],
+              placeholderClassName
+            )}
+          >
+            {!isCompleted &&
+              items.map((item, index) => {
+                return (
+                  item.id === activeId && (
+                    <div
+                      key={index}
+                      className={styles[`${baseClass}__placeholder`]}
+                    >
+                      {item.placeholder}
+                    </div>
+                  )
+                );
+              })}
```

Comment:
iterating through the whole array to display just one selected element is not the best idea. You can get this element by its Id and show it here